backend/src/main/java/ai/verta/modeldb/batchProcess/CollaboratorResourceMigration.java

```diff
@@ -0,0 +1,186 @@
+        } else {
+          LOGGER.warn("CollaboratorResourceMigration Exception: ", ex);
+        }
+      } else {
+        LOGGER.warn("CollaboratorResourceMigration Exception: ", ex);
+      }
+    } finally {
+      ModelDBUtils.unregisteredBackgroundUtilsCount();
+    }
+
+    LOGGER.info("Migration End");
+  }
+
+  private void migrateProjects() {
+    LOGGER.debug("Projects migration started");
+    Long count = getEntityCount(ProjectEntity.class);
+
+    int lowerBound = 0;
+    final int pagesize = this.paginationSize;
+    LOGGER.debug("Total Projects {}", count);
+
+    while (lowerBound < count) {
+
+      try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
+        CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+
+        CriteriaQuery<ProjectEntity> criteriaQuery =
+            criteriaBuilder.createQuery(ProjectEntity.class);
+        Root<ProjectEntity> root = criteriaQuery.from(ProjectEntity.class);
+
+        CriteriaQuery<ProjectEntity> selectQuery =
+            criteriaQuery.select(root).orderBy(criteriaBuilder.asc(root.get("id")));
+
+        TypedQuery<ProjectEntity> typedQuery = session.createQuery(selectQuery);
+
+        typedQuery.setFirstResult(lowerBound);
+        typedQuery.setMaxResults(pagesize);
+        List<ProjectEntity> projectEntities = typedQuery.getResultList();
+
+        if (projectEntities.size() > 0) {
+          Set<String> userIds = new HashSet<>();
+          for (ProjectEntity projectEntity : projectEntities) {
+            userIds.add(projectEntity.getOwner());
+          }
+          LOGGER.debug("Project userId list : " + userIds);
+
+          // Fetch the experiment owners userInfo
+          Map<String, UserInfo> userInfoMap =
+              authService.getUserInfoFromAuthServer(userIds, null, null);
+          for (ProjectEntity project : projectEntities) {
+            try {
+              WorkspaceDTO workspaceDTO =
+                  roleService.getWorkspaceDTOByWorkspaceId(
+                      userInfoMap.get(project.getOwner()),
+                      project.getWorkspace(),
+                      project.getWorkspace_type());
+              // if projectVisibility is not equals to ResourceVisibility.ORG_SCOPED_PUBLIC then
+              // ignore the CollaboratorType
+              roleService.createWorkspacePermissions(
+                  workspaceDTO.getWorkspaceName(),
+                  project.getId(),
+                  Optional.of(Long.parseLong(project.getOwner())),
+                  ModelDBServiceResourceTypes.PROJECT,
+                  CollaboratorTypeEnum.CollaboratorType.READ_ONLY,
+                  getResourceVisibility(
+                      Optional.ofNullable(
+                          WorkspaceTypeEnum.WorkspaceType.forNumber(project.getWorkspace_type())),
+                      VisibilityEnum.Visibility.forNumber(project.getProject_visibility())));
```

Comment:
I think this getter project.getProject_visibility() will return a transient value (in this case always PRIVATE), but expected is a value from the database

Reply:
No @evgeniilezhnin , `project.getProject_visibility()` return initeger which exists on database. the transient value returns by the `project.getProjectVisibility()`

Reply:
got it